lib/posix/clock.c

```diff
@@ -195,3 +196,32 @@ int gettimeofday(struct timeval *tv, void *tz)
 
 	return res;
 }
+
+/**
+ * @brief Get current real time.
+ *
+ * See IEEE 1003.1
+ */
+int clock_getcpuclockid(pid_t pid, clockid_t *clock_id)
+{
+	/* We don't allow any process ID but our own.  */
+	if (pid != 0 && pid != getpid())
+		return EPERM;
+
+#ifdef CLOCK_PROCESS_CPUTIME_ID
+	*clock_id = CLOCK_PROCESS_CPUTIME_ID;
+	return 0;
+#endif
+
+#ifdef CLOCK_MONOTONIC
+	*clock_id = CLOCK_MONOTONIC;
+	return 0;
+#endif
+
+#ifdef CLOCK_REALTIME
+	*clock_id = CLOCK_REALTIME;
+	return 0;
+#endif
```

Comment:
I think that these few lines can probably just be removed, since it would mainly be the `CLOCK_PROCESS_CPUTIME_ID` that should be returned.